Index: FreeCAD-0.16/src/App/FreeCADInit.py
===================================================================
--- FreeCAD-0.16.orig/src/App/FreeCADInit.py
+++ FreeCAD-0.16/src/App/FreeCADInit.py
@@ -43,6 +43,8 @@ def InitApplications():
 	# Checking on FreeCAD module path ++++++++++++++++++++++++++++++++++++++++++
 	ModDir = FreeCAD.getHomePath()+'Mod'
 	ModDir = os.path.realpath(ModDir)
+	ModShareDir = '/usr/share/freecad/Mod'
+	ModShareDir = os.path.realpath(ModShareDir)
 	BinDir = FreeCAD.getHomePath()+'bin'
 	BinDir = os.path.realpath(BinDir)
 	LibDir = FreeCAD.getHomePath()+'lib'
@@ -68,6 +70,14 @@ def InitApplications():
 		for i in ModDirs: ModDict[i.lower()] = os.path.join(ModDir,i)
 	else:
 		Wrn ("No modules found in " + ModDir + "\n")
+
+	if os.path.isdir(ModShareDir):
+		Log("")
+		ModDirs = os.listdir(ModShareDir)
+		for i in ModDirs: ModDict[i.lower()] = os.path.join(ModShareDir,i)
+	else:
+		Wrn ("No modules found in " + ModShareDir + "\n")
+
 	# Search for additional modules in the home directory
 	if os.path.isdir(HomeMod):
 		HomeMods = os.listdir(HomeMod)
@@ -114,6 +124,8 @@ def InitApplications():
 	sys.path.insert(0,Lib64Dir)
 	sys.path.insert(0,ModDir)
 	Log("Using "+ModDir+" as module path!\n")
+	sys.path.insert(0,ModShareDir)
+	Log("Using "+ModShareDir+" as module path!\n")
 	# new paths must be prepended to avoid to load a wrong version of a library
 	try:
 		os.environ["PATH"] = PathExtension + os.environ["PATH"]
